packages/angular_devkit/build_angular/src/browser/index2.ts

```diff
@@ -102,47 +103,70 @@ export function buildWebpackConfig(
 
   const projectTs = requireProjectModule(getSystemPath(projectRoot), 'typescript') as typeof ts;
 
-  const supportES2015 = tsConfig.options.target !== projectTs.ScriptTarget.ES3
-    && tsConfig.options.target !== projectTs.ScriptTarget.ES5;
-
-  wco = {
-    root: getSystemPath(root),
-    logger: logger.createChild('webpackConfigOptions'),
-    projectRoot: getSystemPath(projectRoot),
-    buildOptions: options,
-    tsConfig,
-    tsConfigPath,
-    supportES2015,
-  };
+  const target = tsConfig.options.target;
+
+  const supportES2015 = target !== projectTs.ScriptTarget.ES3
+    && target !== projectTs.ScriptTarget.ES5;
+
+  const differentialLoading = isDifferentialLoadingNeeded(
+    projectRoot, target || projectTs.ScriptTarget.ES5);
 
-  wco.buildOptions.progress = defaultProgress(wco.buildOptions.progress);
+  const targets = differentialLoading ? [projectTs.ScriptTarget.ES5, target] : [target];
 
-  const webpackConfigs: {}[] = [
-    getCommonConfig(wco),
-    getBrowserConfig(wco),
-    getStylesConfig(wco),
-    getStatsConfig(wco),
-  ];
+  if (differentialLoading) {
+    // For differential loading, the builder needs to created the index.html by itself
+    // without using a webpack plugin.
+    options.index = '';
 
-  if (wco.buildOptions.main || wco.buildOptions.polyfills) {
-    const typescriptConfigPartial = wco.buildOptions.aot
-      ? getAotConfig(wco)
-      : getNonAotConfig(wco);
-    webpackConfigs.push(typescriptConfigPartial);
+    options.esVersionInFileName = true;
   }
 
-  const webpackConfig = webpackMerge(webpackConfigs);
+  return targets.map(target => {
+
+    options.scriptTargetOverride = target;
+
+    wco = {
+      root: getSystemPath(root),
+      logger: logger.createChild('webpackConfigOptions'),
+      projectRoot: getSystemPath(projectRoot),
+      buildOptions: options,
+      tsConfig,
+      tsConfigPath,
+      supportES2015: target !== projectTs.ScriptTarget.ES5 && supportES2015,
+    };
+
+    wco.buildOptions.progress = defaultProgress(wco.buildOptions.progress);
+
+    const webpackConfigs: {}[] = [
+        getCommonConfig(wco),
+        getBrowserConfig(wco),
+        getStylesConfig(wco),
+        getStatsConfig(wco),
+      ];
+
+    if (wco.buildOptions.main || wco.buildOptions.polyfills) {
```

Comment:
It's in `wco.buildOptions.scriptTargetOverride`

Reply:
You are only modifying one instance of the AOT plugin.

The one I highlighted above is untouched hence it will always be build with the same target.

Reply:
Ah, I see. This seems to be related to the new worker features. I think, according to @alexeagle [feedback below](https://github.com/angular/angular-cli/pull/14061#issuecomment-479990716), we should do this in a seperate PR to avoid this one becomes too large.

Do I see this right?